### PR TITLE
DOCS-10930 Salesforce 10.11

### PR DESCRIPTION
**Pull request by sathya0**

### CODE REVIEW

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc [hunk 1]
```diff
@@ -3199,6 +3199,186 @@ Retrieves data from specified objects, whether or not they have been deleted.
...
+| Configuration | String | Name of the configuration to use. | | x
+| Batch info a| <<BatchInfo>> |  BatchInfo to retrieve the results for. |  `#[payload]` | x
+| Content type a| Enumeration, one of:
+
+** XML
+** CSV
+** JSON
+** ZIP_XML
+** ZIP_CSV
+** ZIP_JSON |  Content type used at job creation. If it is not provided, the default value is used. |ContentType.XML  |
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: Global comment:

"If it is not provided, the default value is used." This text is not necessary. This is true all the time.

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc [hunk 2]
```diff
@@ -3199,6 +3199,186 @@ Retrieves data from specified objects, whether or not they have been deleted.
...
+| Configuration | String | Name of the configuration to use. | | x
+| Batch info a| <<BatchInfo>> |  BatchInfo to monitor. |  `#[payload]` |
+| Content type a| Enumeration, one of:
+
+** XML
+** CSV
+** JSON
+** ZIP_XML
+** ZIP_CSV
+** ZIP_JSON |  Content type used at job creation. If it is not provided, the default value is used. |ContentType.XML  |
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: Same comment about the second sentence.

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc [hunk 3]
```diff
@@ -3199,6 +3199,186 @@ Retrieves data from specified objects, whether or not they have been deleted.
...
+* SALESFORCE:MUTUAL_AUTHENTICATION_FAILED
+* SALESFORCE:NOT_FOUND
+* SALESFORCE:INVALID_INPUT
+
+[[queryResultStreamById]]
+=== Query Result Stream By Id
+`<salesforce:query-result-stream-by-id>`
+
+
+Returns an InputStream with the query results for a specific result ID of a submitted BatchInfo. You will need to obtain the result IDs first by using the <<queryResultList>> operation.
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: "You will need to obtain" > "Obtain the result IDs first..." or "You must obtain the result IDs first.."

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc [hunk 4]
```diff
@@ -3199,6 +3199,186 @@ Retrieves data from specified objects, whether or not they have been deleted.
...
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Batch info a| <<BatchInfo>> |  The BatchInfo that returned the result ID. |  `#[payload]` | x
+| Result id | String | The result ID obtained with <<queryResultList>>. | | x
+| Output Mime Type a| String | MIME type of the payload output of this operation.  |  |
+| Output Encoding a| String | Encoding of the payload output of this operation. |  |
+| Streaming Strategy a| * <<repeatable-in-memory-stream>>
+* <<repeatable-file-store-stream>>
+* <<non-repeatable-stream>> |  Configure how Mule processes streams with streaming strategies. Repeatable streams are the default behavior. |  |
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: From the spreadsheet: "Configures how Mule processes streams. Repeatable streams are the default behavior."

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc [hunk 5]
```diff
@@ -3199,6 +3199,186 @@ Retrieves data from specified objects, whether or not they have been deleted.
...
+* SALESFORCE:MUTUAL_AUTHENTICATION_FAILED
+* SALESFORCE:NOT_FOUND
+* SALESFORCE:INVALID_INPUT
+
+[[queryResultStream]]
+=== Query Result Stream (Deprecated)
+`<salesforce:query-result-stream>`
+
+
+Returns an InputStream with the query results of a submitted BatchInfo. Internally, the InputStreams contained in the sequence are requested on-demand (lazy-loading). They should be used only with query jobs.
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: "They should be used only" > "Use them only..."

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc [hunk 6]
```diff
@@ -3199,6 +3199,186 @@ Retrieves data from specified objects, whether or not they have been deleted.
 * SALESFORCE:NOT_FOUND
 * SALESFORCE:INVALID_INPUT
 
+[[queryResultList]]
+=== Query Result List
+`<salesforce:query-result-list>`
+
+Returns a list of result IDs for a specific batch. The Salesforce API documentation states that if the batch requires retries, there will be more than one <result> element in the output. This operation should be used with <<queryResultStreamById>> to obtain the results of the query.
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: "The Salesforce API documentation states" text is not necessary. I suggest:
"If the batch requires retries, there will be more than one <result> element in the output."

"This operation should be used with" > "Use this operation with..."

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc (hunk 3 above)
Comment (dhirsch1) [reply salesforce/10.11/modules/ROOT/pages/salesforce-connector-reference.adoc]: As you know, we are trying to use the operation names that appear in Studio wherever we can, so <<queryResultList>> would be *Query Result List*.

File: salesforce/10.11/modules/ROOT/pages/salesforce-connector-upgrade-migrate.adoc [hunk 7]
```diff
@@ -2,19 +2,23 @@
...
-|10.x |10.10
+|9.x |10.11
+|10.x |10.11
 |===
 
 == Changes
 
+=== 10.11
+
+* The *Query Result Stream* operation is deprecated. There are two new operations, *Query Result List* and *Query Result Stream By Id*, that should be used instead.
```
Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-upgrade-migrate.adoc]: Make active - "Use the new *Query Result List* and *Query Result Stream By Id* operations instead"

Comment (dhirsch1) [inline salesforce/10.11/modules/ROOT/pages/salesforce-connector-upgrade-migrate.adoc]: If you have time, you might want to consider putting each new operation in a bullet and then having a short description of each one (you can probably lift the first line from the Reference).  Something like this:

Use one of the following new operations instead:

* *Query Result List* to xxx 
*  *Query Result Stream by Id* to xxxx

This is not a must do.

**Review by dhirsch1 — DISMISSED** [review]

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**